generators/quarkus/generator.js

```diff
@@ -74,6 +80,23 @@ export default class extends BaseApplicationGenerator {
 
     get [BaseApplicationGenerator.LOADING]() {
         return this.asLoadingTaskGroup({
+            async loadCatalog({ application }) {
+                if (application.buildTool === 'gradle') {
+                    this.loadJavaDependenciesFromGradleCatalog(application.javaDependencies);
+                }
+
+                if (application.buildTool === 'maven') {
+                    const pomFile = this.readTemplate('../../quarkus/resources/pom.xml')?.toString();
+                    const applicationJavaDependencies = this.prepareDependencies(
+                        {
+                            ...getPomVersionProperties(pomFile),
+                        },
+                        'java',
+                    );
+
+                    Object.assign(application.javaDependencies, applicationJavaDependencies);
+                }
+            },
```

Comment:
You should use a pom.xml or gradle catalog, you don't need to use different versions conditionally on build tool.

Reply:
You mean like it the change I pushed?

It does work, but when there is a dependency in both Gradle and Maven with the same name, the version from pom.xml would always be used. e.g. `commons-vfs2` exists in both. I don't think if it's safe to assume that the versions will always be the same. This could be confusing.

Reply:
drop from any of it

Reply:
I'm always adding only a gradle catalog to manage gradle and maven dependencies to new implementations.

Reply:
So, to summarize there are three options to load the versions:

1. Load either pom.xml or libs.versions.toml based on `buildTool`. As in 995b5a243954adab28975f3d03ce1232817c73fa (current state)
2. Load both and then the second loaded one might overwrite some version of the first loaded one. As in f8a886a46ca8f7e37929bb964570d09784892e4e
3. Add all needed versions to **either** pom.xml or libs.versions.toml and just load that for both.

There are quite a few dependencies that are needed for Maven and are missing in Gradle. Especially all the plugin versions as these are Maven specific. I think it wouldn't be beneficial for readability to put all them in one as in 3. But if you say that's the better approach I'll trust your experience. I prefer 1 (current state).

Regarding the rest of the changes they are in my opinion ready for a PR after cleaning up the commits.

Thank you very much for advice.